cookbook/src/tx_segwit-v0.md

```diff
@@ -0,0 +1,351 @@
+# Constructing and Signing Transactions - SegWit V0
+
+In this section, we will construct a [SegWit V0 transaction](https://bitcoinops.org/en/topics/segregated-witness/).
+This is the most common type of transaction on the Bitcoin network today[^today].
+
+This is the `cargo` commands that you need to run this example:
+
+```bash
+cargo add bitcoin --features "std, rand-std"
+```
+
+First we'll need to import the following:
+
+```rust
+use std::str::FromStr;
+
+use bitcoin::hashes::Hash;
+use bitcoin::locktime::absolute;
+use bitcoin::secp256k1::{rand, Message, Secp256k1, SecretKey, Signing};
+use bitcoin::sighash::{EcdsaSighashType, SighashCache};
+use bitcoin::{
+    Address, Network, OutPoint, ScriptBuf, Sequence, Transaction, TxIn, TxOut, Txid, WPubkeyHash,
+    Witness,
+};
+```
+
+Here is the logic behind these imports:
+
+- `std::str::FromStr` is used to parse strings into Bitcoin primitives
+- `bitcoin::hashes::Hash` is used to hash data
+- `bitcoin::locktime::absolute` is used to create a locktime
+- `bitcoin::secp256k1::{rand, Message, Secp256k1, SecretKey, Signing}` is used to sign transactions
+- `bitcoin::sighash::{EcdsaSighashType, SighashCache}` is used to create sighashes
+- `bitcoin::{Address, Network, OutPoint, ScriptBuf, Sequence, Transaction, TxIn, TxOut, Txid, WPubkeyHash, Witness}` is used to construct transactions
+
+Next, we define the following constants:
+
+```rust
+const DUMMY_UTXO_AMOUNT: u64 = 20_000_000;
+const SPEND_AMOUNT: u64 = 5_000_000;
+const CHANGE_AMOUNT: u64 = 14_999_000; // 1000 sat fee.
+```
+
+- `DUMMY_UTXO_AMOUNT` is the amount of the dummy UTXO we will be spending
+- `SPEND_AMOUNT` is the amount we will be spending from the dummy UTXO
+- `CHANGE_AMOUNT`[^change] is the amount we will be sending back to ourselves as change
+
+Before we can construct the transaction, we need to define some helper functions[^expect]:
+
+```rust
+# use bitcoin::secp256k1::{rand, Secp256k1, SecretKey, Signing};
+# use bitcoin::WPubkeyHash;
+fn senders_keys<C: Signing>(secp: &Secp256k1<C>) -> (SecretKey, WPubkeyHash) {
+    let sk = SecretKey::new(&mut rand::thread_rng());
+    let pk = bitcoin::PublicKey::new(sk.public_key(secp));
+    let wpkh = pk.wpubkey_hash().expect("key is compressed");
+
+    (sk, wpkh)
+}
+```
+
+`senders_keys` generates a random private key and derives the corresponding public key hash.
+This will be useful to mock a sender.
+In a real application these would be actual secrets.
+Under the hood we are using the [`secp256k1`](https://github.com/rust-bitcoin/rust-secp256k1/) crate to generate the key pair.
+`rust-secp256k1` is a wrapper around [libsecp256k1](https://github.com/bitcoin-core/secp256k1), a C
```

Comment:
nit: Please don't bother changing unless there are other things that you are changing anyways.

This mention of secp and rust-secp seems like a good candidate for a footnote  and not directly relevant to the code.

Reply:
Thanks for the suggestion. Done in f89936122927a1add2f309316ee924c644329c85

Reply:
Both in Segwit and Taproot